plugins/rule_engines/irods_rule_language/src/configuration.cpp

```diff
@@ -33,42 +33,7 @@
 using log_re = irods::experimental::log::rule_engine;
 constexpr auto err_buf_len = ERR_MSG_LEN * 1024;
 
-Cache::Cache() : address(NULL), /* unsigned char *address */
-    pointers(NULL), /* unsigned char *pointers */
-    dataSize(0), /* size_t dataSize */
-    cacheSize(0), /* size_t cacheSize */
-    coreRuleSetStatus(UNINITIALIZED), /* RuleEngineStatus coreRuleSetStatus */
-    appRuleSetStatus(UNINITIALIZED), /* RuleEngineStatus appRuleSetStatus */
-    extRuleSetStatus(UNINITIALIZED), /* RuleEngineStatus extRuleSetStatus */
-    sysFuncDescIndexStatus(UNINITIALIZED), /* RuleEngineStatus sysFuncDescIndexStatus */
-    coreFuncDescIndexStatus(UNINITIALIZED), /* RuleEngineStatus coreFuncDescIndexStatus */
-    appFuncDescIndexStatus(UNINITIALIZED), /* RuleEngineStatus appFuncDescIndexStatus */
-    extFuncDescIndexStatus(UNINITIALIZED), /* RuleEngineStatus extFuncDescIndexStatus */
-    ruleEngineStatus(UNINITIALIZED), /* RuleEngineStatus ruleEngineStatus */
-    cacheStatus(UNINITIALIZED), /* RuleEngineStatus cacheStatus */
-    sysRegionStatus(UNINITIALIZED), /* RuleEngineStatus sysRegionStatus */
-    coreRegionStatus(UNINITIALIZED), /* RuleEngineStatus regionCoreStatus */
-    appRegionStatus(UNINITIALIZED), /* RuleEngineStatus regionAppStatus */
-    extRegionStatus(UNINITIALIZED), /* RuleEngineStatus regionExtStatus */
-    coreRuleSet(NULL), /* RuleSet *coreRuleSet */
-    appRuleSet(NULL), /* RuleSet *appRuleSet */
-    extRuleSet(NULL), /* RuleSet *extRuleSet */
-    sysFuncDescIndex(NULL), /* Env *sysFuncDescIndex */
-    coreFuncDescIndex(NULL), /* Env *coreFuncDescIndex */
-    appFuncDescIndex(NULL), /* Env *appFuncDescIndex */
-    extFuncDescIndex(NULL), /* Env *extFuncDescIndex */
-    sysRegion(NULL), /* Region *sysRegion */
-    coreRegion(NULL), /* Region *coreRegion */
-    appRegion(NULL), /* Region *appRegion */
-    extRegion(NULL), /* Region *extRegion */
-    tvarNumber(0), /* int tvarNumber */
-    clearDelayed(0), /* int clearDelayed */
-    timestamp(time_type_initializer), /* time_type timestamp */
-    logging(0), /* int logging */
-    ruleBase(""), /* char ruleBase[RULE_SET_DEF_LENGTH] */
-    hash("") /* char *hash */{}
-
-Cache ruleEngineConfig;
+Cache ruleEngineConfig{};
```

Comment:
Consider putting a `NOLINT` here to satisfy clang-tidy. Bonus points if you leave an explanatory comment :) OR, don't, so it's not hidden from view. Up to you.

Reply:
Hmm, is Cache a `const` candidate? Can you `memset(...)` a const struct (or is that confusing/bad practice)? I forget...

If you can, we should probably just do that now.
If you can't, we can add the ignore.

Reply:
I see many places where `ruleEngineConfig` is being updated, so I don't think this should be `const`.

Reply:
I'll add an ignore and a comment shortly then.

Reply:
Done.